go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -1321,7 +1320,7 @@ func overwriteRoutesTableID(routes []netlink.Route, tableID int) {
 }
 
 func getRouteTableID(ifIndex int) int {
-	return ifIndex + routingTableIDStart
+	return ifIndex + int(ovnconfig.UserDefinedNetworks.VRFTableBase) + int(ovnconfig.UserDefinedNetworks.MaxNetworks)
```

Comment:
We should check network segmentation feature gate and return `routingTableIDStart` otherwise.